ledfx/effects/slide.py

```diff
@@ -0,0 +1,89 @@
+from ledfx.effects.audio import AudioReactiveEffect, FREQUENCY_RANGES_SIMPLE, MIN_MIDI, MAX_MIDI
+from ledfx.effects.gradient import GradientEffect
+from ledfx.effects.math import ExpFilter, TemporalFilter
+from ledfx.effects import mix_colors
+from ledfx.color import COLORS
+from ledfx.events import GraphUpdateEvent
+import voluptuous as vol
+import numpy as np
+import aubio
+import math
+
+class Slide2AudioEffect(AudioReactiveEffect, GradientEffect):
+
+    NAME = "Slide2"
+
+    CONFIG_SCHEMA = vol.Schema({
+        vol.Optional('blur', description='Amount to blur the effect', default = 1.0): vol.Coerce(float),
+        vol.Optional('mirror', description='Mirror the effect', default = True): bool,
+        vol.Optional('resolution', description='Number of colour bands', default = 7): vol.All(vol.Coerce(int), vol.Range(min=3, max=10)),
+        #vol.Optional('fade_rate', description='Rate at which notes fade', default = 0.15):  vol.All(vol.Coerce(float), vol.Range(min=0.0, max=1.0)),
+        vol.Optional('responsiveness', description='Responsiveness to the note changes', default = 0.03):  vol.All(vol.Coerce(float), vol.Range(min=0.01, max=1.0)),
+    })
+
+    def config_updated(self, config):
+        self.bands = np.zeros(self._config["resolution"])
+        self.band_smoothing = ExpFilter(np.tile(1e-1, self._config["resolution"]), alpha_decay=0.05, alpha_rise=0.9)
+        self.temporal_filter = None
+
+
+    def audio_data_updated(self, data):
+        # TODO move to config_updated when self.pixel_count at that point
+        if not self.temporal_filter:
+            self.temporal_filter = TemporalFilter(self.pixels, rise=self._config["responsiveness"], decay=0.1)
+
+        # Populate octave and note data
+        data.melbank()
+        # Get octave and note data
+        octave = data.averaged_octave
+        # Split octave into bins, find max value per bin
+        maxed_bins = [i.max() for i in np.array_split(octave, self._config["resolution"])]
+        # Update colour bands with max from each octave bin
+        self.bands = self.band_smoothing.update(maxed_bins)
+
+        # TODO move to config_updated, when pixel count is known during effect setup
+        # Get colours from band indexes (scaled between 0 and 1)
+        self.colours = np.array([self.get_gradient_color(i) for i in np.linspace(0, 1, self._config["resolution"])]).astype(int)
```

Comment:
This should be moved in init. Could also consider using _generate_bezier_curve instead which is more performant